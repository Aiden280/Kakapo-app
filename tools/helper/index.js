import fs from 'fs-extra';

// Upload stats.json to `http://webpack.github.io/analyse` for analysis.
// Also you can analyze size with `https://github.com/robertknight/webpack-bundle-size-analyzer`:
// `cat stats.json | analyze-bundle-size`
export const writeStats = stats =>
  fs.writeFile('stats.json', JSON.stringify(stats.toJson(), null, 4));

export default {};
