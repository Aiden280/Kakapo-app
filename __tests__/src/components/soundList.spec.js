import React from 'react';
import { shallow } from 'enzyme';
import test from 'tape';
import { inc, set, lensProp } from 'ramda';
import { newSoundObj } from 'utils/';
import { SoundList } from 'components/';
import { getData } from '../helper';

function setup(props = {}) {
  const propData = {
    ...getData('sounds'),
    ...getData('themes'),
    ...getData('intl'),
    ...props
  };
  return { props, wrapper: shallow(<SoundList {...propData} />) };
}

function randomSounds(count) {
  let arr = new Map();
  for (let i = 0; i < count; inc(i)) {
    const obj = { ...newSoundObj, progress: i > 2 ? 1 : 0.5, editing: i > 2 };
    arr = set(
      lensProp(i),
      Object.keys(obj).reduce((newObj, _e) => {
        newObj[_e] = obj[_e] === null ? `test${i}` : obj[_e];
        return newObj;
      }, {}),
      arr
    );
  }
  return arr;
}

test('<SoundList/> render empty', t => {
  t.plan(1);
  const { wrapper } = setup();
  t.equals(wrapper.html(), '<div></div>');
});

test('<SoundList/> render', t => {
  t.plan(1);
  const { wrapper } = setup({ sounds: randomSounds(4) });
  t.equals(wrapper.type(), 'section');
});

test('<SoundList/> progress === 1', t => {
  t.plan(1);
  const { wrapper } = setup({ sounds: randomSounds(8) });
  t.equals(wrapper.find('.sound-item-wrap').length, 5);
});
