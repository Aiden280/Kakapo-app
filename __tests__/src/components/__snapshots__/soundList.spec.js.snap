// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundList/> progress === 1 1`] = `
<section>
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;

exports[`<SoundList/> render 1`] = `
<section>
  <div
    className="sound-list"
  />
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <label
              htmlFor="name"
            >
              <input
                autoComplete="off"
                defaultValue="thunder"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Name
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <label
              htmlFor="tags"
            >
              <input
                autoComplete="off"
                defaultValue={null}
                name="tags"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              Tag
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            Cancel
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;

exports[`<SoundList/> render empty 1`] = `<div />`;
